src/battlegoals/partyBattleGoals.tsx

```diff
@@ -41,22 +38,45 @@ export default class PartyBattleGoals extends React.Component<NoProps, PartyBatt
       includeSatireGaming: false,
       drawnBattleGoals: [],
       currentPlayer: undefined,
-      picks: {},
+      picks: new Map(),
       hover: -1
     }
   }
 
+  private storeState() {
+    const dto = {
+      includeSatireGaming: this.state.includeSatireGaming,
+      includeOfficial: this.state.includeOfficial,
+      battleGoalIds: this.state.drawnBattleGoals.map(it => it.globalCardId.asString()),
+      picks: [...this.state.picks].map(it => [it[0], it[1].asString()]).map(it => `${it[0]}${PartyBattleGoals.pickStorageSeparator}${it[1]}`)
```

Comment:
I dont' understand this code... :)

Reply:
yes, this was not as straight forward as I would have liked.
I want to have an explicit storage format and not rely on 'just' stringify the domain Object `CardIdentifier`